services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -417,95 +411,57 @@ const cacheLegacyAccountInfo = async () => {
-			} else {
-				// Calculate number of batches that already have been indexed and continue
-				const prevGenesisAccountIndexingBatchNum = genesisAccountIndexingBatchNum;
-				genesisAccountIndexingBatchNum = Math.floor(numAccountsIndexed / BATCH_SIZE);
-				if (indexedTillBatch === undefined || prevGenesisAccountIndexingBatchNum === 0) {
-					logger.info(`Genesis accounts already indexed until batch: ${genesisAccountIndexingBatchNum - 1}, will continue from batch ${genesisAccountIndexingBatchNum}`);
-				}
-			}
+	const genesisAccountsToIndex = genesisBlock.asset.accounts;
+	const genesisAccountPageCached = 'genesisAccountPageCached';
 
-			const batchNum = genesisAccountIndexingBatchNum; // Use shorter alias
-			const genesisAccountAddressesToIndex = genesisAccountsToIndex
-				.slice(batchNum * BATCH_SIZE, (batchNum + 1) * BATCH_SIZE)
-				.map(account => account.address);
+	logger.info(`${genesisAccountsToIndex.length} registered accounts found in the genesis block`);
 
-			logger.debug(`numAccountsIndexed: ${numAccountsIndexed}, numGenesisAccounts: ${genesisAccountsToIndex.length}`);
+	const lastCachedPage = await genesisAccountsCache.get(genesisAccountPageCached) || 0;
 
-			if (numAccountsIndexed >= genesisAccountsToIndex.length) {
-				logger.info(`Genesis block accounts already indexed from height ${genesisHeight}`);
-				Signals.get('newBlock').remove(indexGenesisAccounts);
-			} else {
-				if (batchNum === 0) {
-					logger.info(`Starting indexing of genesis block accounts from height ${genesisHeight} in batches of ${BATCH_SIZE}`);
-
-					// Index the genesis block accounts
-					const initDelegateAddresses = genesisBlock.asset.initDelegates;
-					await indexAccountsbyAddress(initDelegateAddresses, true);
-				}
-
-				logger.info(`Indexing genesis account batch: ${batchNum}, ${Math.ceil(genesisAccountsToIndex.length / BATCH_SIZE) - batchNum - 1} to go`);
-
-				const PAGE_SIZE = 1000;
-				const NUM_PAGES = Math.ceil(genesisAccountAddressesToIndex.length / PAGE_SIZE);
-				for (let i = 0; i < NUM_PAGES; i++) {
-					// eslint-disable-next-line no-await-in-loop
-					await indexAccountsbyAddress(
-						genesisAccountAddressesToIndex.slice(i * PAGE_SIZE, (i + 1) * PAGE_SIZE),
-						true,
-					);
-				}
+	const PAGE_SIZE = 1000;
+	const NUM_PAGES = Math.ceil(genesisAccountsToIndex.length / PAGE_SIZE);
+	for (let pageNum = 0; pageNum < NUM_PAGES; pageNum++) {
+		const currentPage = pageNum * PAGE_SIZE;
+		const nextPage = (pageNum + 1) * PAGE_SIZE;
+		const percentage = Math.round(((nextPage - 1) / genesisAccountsToIndex.length) * 1000);
+
+		if (pageNum >= lastCachedPage) {
+			const slicedAccounts = genesisAccountsToIndex.slice(currentPage, nextPage);
+			const genesisAccountAddressesToIndex = slicedAccounts
+				.filter(account => account.address.length === 40)
+				.map(account => account.address);
+	
+			logger.info(`Scheduling retrieval of genesis accounts batch ${pageNum}/${NUM_PAGES} (${(percentage / 10).toFixed(1)}%)`);
```

Comment:
The final log ends with a percentage value of `100.2%` against the Mainnet and batch number `308/309` instead of `309/309` at `100.0%`. For clarity in the logs please consider the following change, as also suggested by @nagdahimanshu 

Also, for cosmetic purposes, please consider logging batch numbers as `${pageNum + 1}/${NUM_PAGES}` instead of `${pageNum}/${NUM_PAGES}`.

```
2021-09-21T15:48:34.353 INFO [blocks] Scheduling retrieval of genesis accounts batch 307/309 (99.9%)
2021-09-21T15:48:35.291 INFO [blocks] Scheduling retrieval of genesis accounts batch 308/309 (100.2%)
```

```suggestion
		const percentage = (((pageNum + 1) / NUM_PAGES) * 100).toFixed(1);

		if (pageNum >= lastCachedPage) {
			const slicedAccounts = genesisAccountsToIndex.slice(currentPage, nextPage);
			const genesisAccountAddressesToIndex = slicedAccounts
				.filter(account => account.address.length === 40)
				.map(account => account.address);
	
			logger.info(`Scheduling retrieval of genesis accounts batch ${pageNum + 1}/${NUM_PAGES} (${percentage}%)`);
```